DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/CryptoHashProcessor.cs

```diff
@@ -0,0 +1,88 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Text;
+using Dicom;
+using Dicom.IO.Buffer;
+using EnsureThat;
+using Microsoft.Health.DeID.SharedLib.Settings;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.DeID.SharedLib;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class CryptoHashProcessor : IAnonymizerProcessor
+    {
+        private readonly CryptoHashFunction _cryptoHashFunction;
+
+        public CryptoHashProcessor(JObject settingObject, IAnonymizerSettingsFactory settingFactory = null)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            settingFactory ??= new AnonymizerSettingsFactory();
+            var cryptoHashSetting = settingFactory.CreateAnonymizerSetting<CryptoHashSetting>(settingObject);
+            _cryptoHashFunction = new CryptoHashFunction(cryptoHashSetting);
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            if (item is DicomStringElement)
```

Comment:
> if (item is DicomStringElement) [](http://example.com/codeflow?start=11&length=32)

just curious, does a switch block look better to you? :-)

Reply:
Switch may involve some problems.
The scope of  variables is the entire switch expression, I can not use same variable name (e.g. encryptedValues) in different cases even they have same meaning but different types.

Reply:
I thinks switch is better when too much if-else.